src/shared/utilities/vsCodeUtils.ts

```diff
@@ -71,3 +71,12 @@ export async function activateExtension<T>(
 export function promisifyThenable<T>(thenable: Thenable<T>): Promise<T> {
     return new Promise((resolve, reject) => thenable.then(resolve, reject))
 }
+
+export function isUntitledScheme(uri: vscode.Uri): boolean {
+    return uri.scheme === 'untitled'
+}
+
+// If the VSCode URI is untitled then return the string representation, otherwise normalize the filesystem path
+export function normalizeVSCodeUri(uri: vscode.Uri): string {
+    return isUntitledScheme(uri) ? uri.toString() : pathutils.normalize(uri.fsPath)
```

Comment:
should we make a stronger check in the normalize() case, to verify that this is indeed a filesystem Uri?